gcloud-java-dns/src/test/java/com/google/cloud/dns/it/ITDnsTest.java

```diff
@@ -967,25 +969,920 @@ public void testListDnsRecords() {
+      assertEquals(0, retrieved.deletions().size());
+      assertNull(retrieved.startTimeMillis());
+      assertNull(retrieved.status());
+      retrieved = statusResult.get();
+      assertEquals(created.generatedId(), retrieved.generatedId());
+      assertEquals(0, retrieved.additions().size());
+      assertEquals(0, retrieved.deletions().size());
+      assertNull(retrieved.startTimeMillis());
+      assertEquals(ChangeRequestInfo.Status.DONE, retrieved.status());
+      retrieved = timeResult.get();
+      assertEquals(created.generatedId(), retrieved.generatedId());
+      assertEquals(0, retrieved.additions().size());
+      assertEquals(0, retrieved.deletions().size());
+      assertEquals(created.startTimeMillis(), retrieved.startTimeMillis());
+      assertNull(retrieved.status());
+      retrieved = additionsResult.get();
+      assertEquals(created.generatedId(), retrieved.generatedId());
+      assertEquals(2, retrieved.additions().size());
+      assertTrue(retrieved.additions().contains(A_RECORD_ZONE1));
+      assertTrue(retrieved.additions().contains(AAAA_RECORD_ZONE1));
+      assertEquals(0, retrieved.deletions().size());
+      assertNull(retrieved.startTimeMillis());
+      assertNull(retrieved.status());
+      // finishes with delete otherwise we cannot delete the zone
+      created = zone.applyChangeRequest(CHANGE_DELETE_ZONE1,
+          Dns.ChangeRequestOption.fields(ChangeRequestField.DELETIONS));
+      batch = DNS.batch();
+      DnsBatchResult<ChangeRequest> deletionsResult =
+          batch.getChangeRequest(zone.name(), created.generatedId(),
+              Dns.ChangeRequestOption.fields(ChangeRequestField.DELETIONS));
+      batch.submit();
+      retrieved = deletionsResult.get();
+      assertEquals(created.generatedId(), retrieved.generatedId());
+      assertEquals(0, retrieved.additions().size());
+      assertEquals(2, retrieved.deletions().size());
+      assertTrue(retrieved.deletions().contains(AAAA_RECORD_ZONE1));
+      assertTrue(retrieved.deletions().contains(A_RECORD_ZONE1));
+      assertNull(retrieved.startTimeMillis());
+      assertNull(retrieved.status());
+      waitForChangeToComplete(zone.name(), created.generatedId());
+    } finally {
+      clear();
+    }
+  }
+
+  @Test
+  public void testListChangesBatch() {
+    try {
+      DnsBatch batch = DNS.batch();
+      DnsBatchResult<Page<ChangeRequest>> result = batch.listChangeRequests(ZONE1.name());
+      batch.submit();
+      try {
+        result.get();
+        fail("Zone does not exist yet");
+      } catch (DnsException ex) {
+        // expected
+        assertEquals(404, ex.code());
+        assertFalse(ex.retryable());
+      }
+      // zone exists but has no changes
+      DNS.create(ZONE1);
+      batch = DNS.batch();
+      result = batch.listChangeRequests(ZONE1.name());
+      batch.submit();
+      ImmutableList<ChangeRequest> changes = ImmutableList.copyOf(result.get().iterateAll());
```

Comment:
Is it necessary to copy the iterable into a list? You could use [Iterables.size](http://docs.guava-libraries.googlecode.com/git/javadoc/com/google/common/collect/Iterables.html#size(java.lang.Iterable))